packages/react-components/react-table/src/components/TableHeaderCell/useTableHeaderCell.tsx

```diff
@@ -56,7 +56,7 @@ export const useTableHeaderCell_unstable = (
         }),
       },
     }),
-    sortable,
+    sortable: props.sortable || sortable,
```

Comment:
I would keep this as 
```suggestion
    sortable,
```
modify the context call like this:
```ts
const { noNativeElements, sortable: contextSortable } = useTableContext();
const { sortable = contextSortable } = props;
```